bundles/core/src/main/java/com/adobe/cq/commerce/core/components/utils/SiteNavigation.java

```diff
@@ -166,4 +189,81 @@ public static String toProductUrl(String pagePath, String slug, String variantSk
         String variantSku = StringUtils.substringAfter(combinedSku, COMBINED_SKU_SEPARATOR);
         return Pair.of(baseSku, variantSku.isEmpty() ? null : variantSku);
     }
+
+    /**
+     * Builds and returns a category or product page URL based on the given page and slug.
+     * This method might return the URL of a specific subpage configured for that particular page.
+     * 
+     * @param page The page used to build the URL.
+     * @param slug The slug of the product or the category.
+     * @return The page URL.
+     */
+    public String toPageUrl(Page page, String slug) {
+        return toProductUrl(page, slug, null);
+    }
+
+    /**
+     * Builds and returns a product page URL based on the given page, slug, and variant sku.
+     * This method might return the URL of a specific subpage configured for that particular page.
+     * 
+     * @param page The page used to build the URL.
+     * @param slug The slug of the product.
+     * @param variantSku An optional sku of the variant that will be "selected" on the product page, can be null.
+     * @return The product page URL.
+     */
+    public String toProductUrl(Page page, String slug, String variantSku) {
```

Comment:
If `SiteNavigation` has the request object then all the methods asking for the `Page` should be refactored to just use the request path from the request object.

Reply:
That would be too easy, links are not created for and to the current page ;-) For example, on the category page you create links to product pages.